### ISSUE TITLE
New cards are created at the bottom of the screen

### ISSUE THREAD

**Issue by sparunakian** [nextcloud/deck#6457]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Describe the bug**
Since Nextcloud 30 and Deck, the create card button in a board creates a text input at the bottom of the screen.

**To Reproduce**
1. Open the Deck app
2. Go to a board
3. Click the create card button ![Image](https://github.com/user-attachments/assets/bbd3c21d-c9b7-49e7-b6d0-380e2e1a6d14)
4. The text input for the new card title is displayed at the bottom of the screen

**Expected behavior**
Previously, the text input was displayed next to the last card in the column.

**Screenshots**
![Image](https://github.com/user-attachments/assets/83c7490f-9deb-405d-947a-b4049e8d5c0e)

**Client details:**
 - OS: ArchLinux & Debian
 - Browser: Firefox
 - Version: 132 & 128.3.1 ESR
 - Device: Desktop

<details>
<summary>Server details</summary>

**Operating system**: Debian & ArchLinux

**Web server:** Apache

**Database:** MySQL

**PHP version:** 8.2

**Nextcloud version:** 30.0.1

**Where did you install Nextcloud from:** Zip archive (Debian) & distribution package (ArchLinux)

**Signing status:**

```
Login as admin user into your Nextcloud and access
http://example.com/index.php/settings/integrity/failed
paste the results here.
```

**List of activated apps:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```

**Nextcloud configuration:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or

Insert your config.php content here
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

</details>

<details>
<summary>Logs</summary>

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```

#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```

</details>

**Comment by luka-nextcloud** [comment]
Confirmed with the design team, we should fix this.

**Comment by XueSheng-GIT** [comment]
@luka-nextcloud thanks for taking care! Is there any quick fix/workaround available? How ever this may look in future, is it possible to revert back to the old behaviour for now so that users are able to actually see their new cards again?

**Comment by ludij** [comment]
Please see the PR above, which is an attempt to improve the responsive behavior and solves this issue as a side effect :)

**Comment by bentuna** [comment]
Any news on this? On iOS via Browser, Deck is practically unusable because you can't see what you type when creating a new card.

https://github.com/user-attachments/assets/33ae97a9-1246-4500-ad21-4d61ee457b13

**Comment by bentuna** [comment]
**Workaround**
We are using the following CSS hack to make the "new card" input field appear **above** older cards - works nicely:

```
.app-deck {
  .stack__card-add form {
    order: 1;
  }
  .smooth-dnd-container.vertical {
    order: 2;
  }
}
```

**Screenshot**

<img width="644" height="446" alt="Image" src="https://github.com/user-attachments/assets/06bc188d-2aa0-4fca-be0e-971af7cf96cc" />

**Comment by ziggycross** [comment]
@bentuna could you submit your updates as a PR? It seems to solve the issue. I would love to use it but am too new to Nextcloud to know where I would need to patch the CSS.

**Comment by bentuna** [comment]
@ziggycross the problem is, that my changes are only CSS hacks which I inject via the [custom CSS app](https://apps.nextcloud.com/apps/theming_customcss) - I don't have the knowledge to fix this upstream. :/

**Comment by xstable** [comment]
I've created a PR. Hope it will be merged.

**Comment by ziggycross** [comment]
It looks like this has been resolved in https://github.com/nextcloud/deck/pull/6640/files#diff-81373dc73b1e925b5c12bf0860a06f0def544c9b35b18dabec6c0548fad1d536R313
Likely a stale ticket.

**Comment by theoholl** [comment]
Still looks like this on `main`:

<img width="2826" height="1518" alt="Image" src="https://github.com/user-attachments/assets/b059f782-6a0b-4131-b4cb-4de5304c7aaa" />

**Comment by theoholl** [comment]
> Confirmed with the design team that this should be fixed.

I agree that this should be fixed, and it’s great that there’s already a PR up for it. I’m mainly wondering which solution is the best fit for the Deck app. I’d prefer the **Add card** button to stay close to where the new card will appear. My thinking is that if the input field is shown above the cards, the new card should be added to the top of the stack; if the input field is shown below the cards, the new card should be added to the bottom.

I've outlined some options below. What is your opinion? @luka-nextcloud @xstable @bentuna, everyone

|  | Option A: Above stack | Option B: Below last card | Option C: Bottom of screen |
|---|---|---|---|
| **'Add card' button** | Placed above the card stack | Placed below the last card | Fixed to the bottom of the screen |
| **Card title input** | Appears above the card stack | Appears below the last card; if the list is long, it stays fixed to the bottom of the screen | Opens from the bottom of the screen |
| **New card placement** | Inserted at the top of the stack | Added after the last card | Added after the last card |
| **Example** |  <img width="1288" height="892" alt="Image" src="https://github.com/user-attachments/assets/b6a22de7-d586-47c8-9b54-b374efc80a9a" /> | <img width="501" height="484" alt="Image" src="https://github.com/user-attachments/assets/55ac7926-9304-4d86-b891-782f49abe5a1" /> | <img width="1600" height="987" alt="Image" src="https://github.com/user-attachments/assets/5c968b37-0d67-4428-a01f-e02ddc649f86" /> |

**Comment by bentuna** [comment]
Thank you very much @theoholl - I completely agree that 'Add card' button, input and placement of the new card should be in the same place - so we have no annoying unneccessary cursor movements. The 'Add card' button becomes the input becomes the new card.

How about combining options A and B and having the possibility to add cards at the top OR the bottom of the stack? 

<img width="644" height="459" alt="Image" src="https://github.com/user-attachments/assets/2816aed4-4935-4778-afcf-cc18334311c5" />

**Comment by theoholl** [comment]
@bentuna , I really like the suggestion! Should we add a toggle to settings to show either the add to top or add to bottom option? Which one should be default?

**Comment by bentuna** [comment]
> Should we add a toggle to settings to show either the add to top or add to bottom option? Which one should be default?

In my opinion global settings are not very useful here because you will always have different needs in different lists - on one list you wanna add on top, on one you wanna add at the bottom, on another you add on both ends.

I think it would be best to make the default to show both buttons and to have two **separate** switches in the settings:
* Hide 'Add card' button on top of lists
* Hide 'Add card' button at the bottom of lists

So people can also hide both buttons (by activating both switches) for example to print or make screenshots and have a clean list view with just the existing cards.

**Comment by bentuna** [comment]
But we should have a additional way of adding cards that works even when you hide both in-list buttons: We could add two entries to the '...' menu of the list header: 
* Add card on top
* Add card at bottom

<img width="644" height="459" alt="Image" src="https://github.com/user-attachments/assets/38a4d0a9-1dbe-4538-b33f-5c2e85e5b20b" />

**Comment by ziggycross** [comment]
You should also consider adding the ability to double click in a column to add a new card.

Also, I find for anything other than the bottom of the list, it is nicer if the add buttons only appear on hover. That way it doesn't reduce the amount of available screen space.

**Comment by bentuna** [comment]
> You should also consider adding the ability to double click in a column to add a new card.
> 
> Also, I find for anything other than the bottom of the list, it is nicer if the add buttons only appear on hover. That way it doesn't reduce the amount of available screen space.

I would argue against that - neither double click nor hover are mobile friendly.

Although I see your point with the screen space. But I think we are addressing that with the suggested settings:
> * Hide 'Add card' button on top of lists
> * Hide 'Add card' button at the bottom of lists

**Comment by aplteam** [comment]
New cards still go to the bottom?!
